### PR TITLE
Skipping tests in evp_test leaks memory

### PR DESCRIPTION
**Pull request by tmshort**
When configured with "no-mdc2 enable-crypto-mdebug" the evp_test
will leak memory due to skipped tests, and error out.

Also fix a skip condition

### CODE REVIEW

File: test/evp_test.c [hunk 1]
```diff
@@ -351,21 +351,22 @@ static int setup_test(struct evp_test *t, const struct evp_test_method *tmeth)
...
+            }
+            if (!check_test_error(t)) {
+                if (t->err)
+                    ERR_print_errors_fp(stderr);
+                t->errors++;
+            }
         }
+        /* clean it up */
         ERR_clear_error();
         t->meth->cleanup(t);
```
Comment (tmshort) [inline test/evp_test.c]: Although, I supposed the following changes to the cleanup routines could be done instead with a simple check for t->data before calling t->meth->cleanup(t)...

**Comment by richsalz** [comment]
@tmshort can you rebase and then I'll merge.  thanks!  ( @levitte I assume you're okay with the rebase, let me know if you want to re-review.

**Comment by tmshort** [comment]
Rebase to where? back onto OpenSSL_1_1_0? master?

**Comment by richsalz** [comment]
Rebase to current master, I'll cherry-pick to 1.1.0-stable, thanks!

**Comment by tmshort** [comment]
https://github.com/openssl/openssl/pull/1968

**Comment by richsalz** [comment]
commit 0a2a5d2 on 1.1.0 thanks.

**Pull request closed without merging**